src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -130,18 +130,15 @@ public static partial class JsonSerializer
                     // Avoid a reference to the JsonNode type for trimming
                     if (jsonPropertyInfo.PropertyType.FullName == JsonTypeInfo.JsonObjectTypeName)
                     {
-                        extensionData = jsonPropertyInfo.EffectiveConverter.CreateObject(options);
+                        ThrowHelper.ThrowInvalidOperationException_NodeJsonObjectCustomConverterNotAllowedOnExtensionProperty();
```

Comment:
Yep. Not strictly necessary in this changed but I figured we might want to consolidate infrastructure concerns rather than polluting `JsonConverter` with a bunch of ad-hoc virtual methods.